### PR TITLE
Added support for Me request

### PR DESCRIPTION
**Pull request by slavikm**
## Description
1. Add support for `/v1/auth/me` request to retrieve current user details
2. Added missing `UserID` field to UserResponse

### CODE REVIEW

File: descope/auth/auth.go [hunk 1]
```diff
@@ -126,6 +126,30 @@ func (auth *authenticationService) LogoutWithOptions(request *http.Request, opti
 	return nil
 }
 
+func (auth *authenticationService) Me(request *http.Request) (*UserResponse, error) {
+	if request == nil {
+		return nil, errors.MissingProviderError
```
Comment (orius123) [inline descope/auth/auth.go:131]: Wrong error

Comment (slavikm) [reply descope/auth/auth.go]: Fixed both here and logout

File: descope/auth/auth.go [hunk 2]
```diff
@@ -126,6 +126,30 @@ func (auth *authenticationService) LogoutWithOptions(request *http.Request, opti
...
 }
 
+func (auth *authenticationService) Me(request *http.Request) (*UserResponse, error) {
+	if request == nil {
+		return nil, errors.MissingRequestError
+	}
+
+	_, refreshToken := provideTokens(request)
+	if refreshToken == "" {
+		logger.LogDebug("unable to find tokens from cookies")
```
Comment (aviadl) [inline descope/auth/auth.go:136]: Not only from cookies, also from header
And yes, I know it is a copy paste, please change the source as well :)

Comment (slavikm) [reply descope/auth/auth.go:136]: I don't think we take `refreshToken` from headers?

Comment (aviadl) [reply descope/auth/auth.go:136]: You are right

**Review by aviadl — APPROVED** [review]

**Pull request merged**